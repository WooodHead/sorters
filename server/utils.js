function prepare(o) {
    if (o && o._id) {
        o._id = o._id.toString()
    }
    return o
}

async function findOne(Collection, ...args) {
    return prepare(await Collection.findOne(...args))
}

async function find(Collection, ...args) {
    return (await Collection.find(...args).toArray()).map(prepare)
}

module.exports = {
    prepare,
    findOne,
    find,
}