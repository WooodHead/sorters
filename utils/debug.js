
export function log(arg) {
    console.log(arg)
    return arg
}
