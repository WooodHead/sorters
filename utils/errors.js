export const errorMessage = (e) => {
    if (e.graphQLErrors && e.graphQLErrors.length > 0) {
        return e.graphQLErrors[0].message
    } else {
        return e.message
    }
}