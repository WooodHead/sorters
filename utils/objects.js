
function toObject(arr) {
    const res = {}
    for (const [key, value] of arr) {
        res[key] = value
    }
    return res
}

function objectMap(object, keyMap = key => key, valueMap = value => value) {
    return Object.keys(object).reduce((o, key) => {
        o[keyMap(key)] = valueMap(object[key])
        return o
    }, {})
}

module.exports = {
    toObject,
    objectMap
}
