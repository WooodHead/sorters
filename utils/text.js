export const replaceAll = (string, search, replacement) => {
  if (typeof string === "string") {
    return string.replace(new RegExp(search, 'g'), replacement)
  } else {
    return string
  }
}

export const teaserAndInfo = (content, min, opt, max) => {
	if (content) {
		if (content.length <= max) {
			return {
				cut: false,
				teaser: content
			};
		}
		const words = content.split(" ");
		let teaser = "";
		for (let i=0; i<words.length; i++) {
			const candidate = words[i];
			let wouldbelength = teaser.length + 1 + candidate.length + 3;
			if (wouldbelength >= opt) {
				// We're done
				if (wouldbelength > max) {
					if (teaser.length + 3 < min) {
						// cut the word
						return {
							teaser: teaser + " " + candidate.substring(0, opt - teaser.length - 1) + "...",
							cut: true,
						};
					} else {
						return {
							teaser: teaser + "...",
							cut: true,
						};
					}
				} else {
					return {
						teaser: teaser + " " + candidate + "...",
						cut: true,
					};
				}
			}
			teaser += " " + candidate;
		}
		return {
			teaser: teaser + "...",
			cut: true,
		};
	} else {
		return {
			teaser: "",
			cut: false,
		};
	}
};
export const teaser = (content, min, opt, max) => {
	const teaser = teaserAndInfo(content, min, opt, max);
	return teaser.teaser;
}
