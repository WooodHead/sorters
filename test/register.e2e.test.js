import cheerio from 'cheerio'
import pretty from 'pretty'
import {ObjectID} from 'mongodb'
import {setup, teardown} from './setup'
import {sleep} from './utils'

let app
let db
let browser

beforeAll(async () => {
    const res = await setup()
    app = res.app
    db = res.db
    browser = res.browser
})

afterEach(async () => {
    await db.dropDatabase()    
})

afterAll(async () => {
    await teardown({db, app, browser})
})

describe('register', () => {
    it('displays', async () => {
        const browserPage = await browser.createPage()
        const status = await browserPage.open(`http://localhost:3000/register`)
        expect(status).toBe('success')

        const text = await browserPage.property('content')
        const $ = cheerio.load(text)
        const page = $('#__next')
        const pageHtml = page.html()
        const html = pretty(pageHtml)
        expect(html).toMatchSnapshot()
    })
})
