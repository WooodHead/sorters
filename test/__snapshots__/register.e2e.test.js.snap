// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`register displays 1`] = `
"<div data-reactroot=\\"\\" data-reactid=\\"1\\" data-react-checksum=\\"-96085406\\">
  <div data-reactid=\\"2\\">
    <!-- react-empty: 3 -->
    <div style=\\"padding-top:50px;padding-bottom:20px;\\" data-reactid=\\"4\\">
      <nav class=\\"navbar navbar-inverse navbar-fixed-top\\" data-reactid=\\"5\\">
        <div class=\\"container\\" data-reactid=\\"6\\">
          <div class=\\"navbar-header\\" data-reactid=\\"7\\">
            <button type=\\"button\\" class=\\"navbar-toggle collapsed\\" data-toggle=\\"collapse\\" data-target=\\"#navbar\\" aria-expanded=\\"false\\" aria-controls=\\"navbar\\" data-reactid=\\"8\\"><span class=\\"sr-only\\" data-reactid=\\"9\\">Toggle navigation</span><span class=\\"icon-bar\\" data-reactid=\\"10\\"></span><span class=\\"icon-bar\\" data-reactid=\\"11\\"></span><span class=\\"icon-bar\\" data-reactid=\\"12\\"></span></button>
            <a class=\\"navbar-brand\\"
              href=\\"/\\" data-reactid=\\"13\\">Sorters Club</a>
          </div>
          <div id=\\"navbar\\" class=\\"collapse navbar-collapse\\" data-reactid=\\"14\\">
            <ul class=\\"nav navbar-nav navbar-right\\" data-reactid=\\"15\\">
              <li class=\\"false\\" data-reactid=\\"16\\">
                <a href=\\"/\\" data-reactid=\\"17\\">Home</a>
              </li>
              <li class=\\"false\\" data-reactid=\\"18\\">
                <a href=\\"/about\\" data-reactid=\\"19\\">About</a>
              </li>
              <li class=\\"false\\" data-reactid=\\"20\\">
                <a href=\\"/news\\" data-reactid=\\"21\\">News</a>
              </li>
              <li class=\\"false\\" data-reactid=\\"22\\">
                <a href=\\"/users\\" data-reactid=\\"23\\">Sorters</a>
              </li>
              <li class=\\"false\\" data-reactid=\\"24\\">
                <a href=\\"/dashboard\\" data-reactid=\\"25\\">Dashboard</a>
              </li>
              <li class=\\"false\\" data-reactid=\\"26\\">
                <a href=\\"/login\\" data-reactid=\\"27\\">Log in</a>
              </li>
              <li class=\\"active\\" data-reactid=\\"28\\">
                <a href=\\"/register\\" data-reactid=\\"29\\">Register</a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
    <div style=\\"max-width:300px;margin:auto;\\" data-reactid=\\"30\\">
      <h1 data-reactid=\\"31\\">Register</h1>
      <form data-reactid=\\"32\\">
        <div class=\\"form-group\\" data-reactid=\\"33\\">
          <label for=\\"email\\" data-reactid=\\"34\\">Email</label>
          <input type=\\"email\\" class=\\"form-control\\" id=\\"email\\" placeholder=\\"Email\\" data-reactid=\\"35\\">
        </div>
        <div class=\\"form-group\\" data-reactid=\\"36\\">
          <label for=\\"password\\" data-reactid=\\"37\\">Password</label>
          <input type=\\"password\\" class=\\"form-control\\" id=\\"password\\" placeholder=\\"******\\" data-reactid=\\"38\\">
        </div>
        <div class=\\"form-group\\" data-reactid=\\"39\\">
          <button type=\\"submit\\" class=\\"btn btn-primary btn-block\\" data-reactid=\\"40\\">Register</button>
        </div>
      </form>
      <p data-reactid=\\"41\\">
        <!-- react-text: 42 -->Already have an account?
        <!-- /react-text -->
        <a href=\\"/login\\" data-reactid=\\"43\\">Log in</a>
        <!-- react-text: 44 -->.
        <!-- /react-text -->
      </p>
    </div>
    <footer data-reactid=\\"45\\">
      <div class=\\"container\\" data-reactid=\\"46\\">
        <hr data-reactid=\\"47\\">
        <p data-reactid=\\"48\\">
          <!-- react-text: 49 -->Brought to you with &#x2764; by
          <!-- /react-text -->
          <a href=\\"/about\\" data-reactid=\\"50\\">Nick Redmark</a>
          <!-- react-text: 51 -->. Support and discuss the development of this platform on
          <!-- /react-text -->
          <a href=\\"https://www.patreon.com/nickredmark\\" target=\\"_blank\\" data-reactid=\\"52\\">Patreon</a>
          <!-- react-text: 53 -->.
          <!-- /react-text -->
        </p>
      </div>
    </footer>
  </div>
</div>"
`;
